deepchem/data/data_loader.py

```diff
@@ -1742,5 +1742,9 @@ def _featurize_shard(self, shard):
             raise ValueError(
                 "Unknown key in yaml file. Please check format for correctness."
             )
-        x = DFTEntry.create(e_type, true_val, systems)
+        if 'weight' in shard.keys():
```

Comment:
In a follow-up PR, let's change this so that it doesn't group multiple datapoints in a shard into one DFTEntry object. A shard with say 10 datapoints should produce 10 objects not 1.

Let's also add a test that we can support shard size > 1.